mcs/class/monodoc/Monodoc.Ecma/EcmaDesc.cs

```diff
@@ -232,20 +232,27 @@ void ConstructCRef (StringBuilder sb)
 			if (DescKind == Kind.Namespace)
 				return;
 
-			sb.Append ('.');
+			if (!string.IsNullOrEmpty (Namespace))
+				sb.Append ('.');
+
 			sb.Append (TypeName);
 			if (GenericTypeArguments != null) {
-				sb.Append ('<');
-				int i=0;
-				foreach (var t in GenericTypeArguments) {
-					if (i > 0) {
-						sb.Append (",");
+				if (GenericTypeArguments.Any (a => a == null)) {
+					sb.AppendFormat ("`{0}", GenericTypeArguments.Count ());
+				}
+				else {
```

Comment:
This `else` block could be avoided -- and the corresponding whitespace changes due to indenting this block -- if the above `if` block used `continue`:

    if (GenericTypeArgumentsIsNumeric) {
        sb.AppendFormat ("`{0}", GenericTypeArgumentsCount);
        continue;
    }
    ...